src/backend/looped/hud/hud_transition_state.cpp

```diff
@@ -103,10 +110,16 @@ namespace big
 			HUD::BUSYSPINNER_OFF();
 		}
 
-		// sometimes when going into a single player mission or transition this one remains on screen permanently
-		if (state == eTransitionState::TRANSITION_STATE_TERMINATE_MAINTRANSITION)
+		if (state == eTransitionState::TRANSITION_STATE_FM_TRANSITION_CREATE_PLAYER)
 		{
-			return;
+			if (g->session.force_session_host && g->session.kick_host_when_forcing_host)
+			{
+				for (auto& [_, plyr] : g_player_service->players())
+				{
+					if (plyr->is_host())
+						kick::lost_connection_kick(plyr);
```

Comment:
Probably needs it a specific point while entering the session to prevent problems with kicking the host too early.

We could maybe switch this to an event-based system but for now as we only have one thing that checks against a specific state it should be fine.